src/components/vm/consoles/vnc.jsx

```diff
@@ -73,12 +75,31 @@ export class VncState extends ConsoleState {
     constructor() {
         super();
         this.sizeMode = "none";
+        this.clean_disconnect = true;
+        this.connection_failure = null;
     }
 
     setSizeMode(val) {
         this.sizeMode = val;
         this.update();
     }
+
+    setConnected(val) {
+        this.connected = val;
+        this.connection_failure = null;
+        this.clean_disconnect = true;
+        this.update();
+    }
+
+    setDisconnected(clean) {
+        this.connected = false;
+        this.clean_disconnect = clean;
+        this.update();
+    }
+
+    setConnectionFailure(val) {
+        this.connection_failure = val;
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2038-7e41038d-20250806-115005-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)